Dart/src/com/jetbrains/lang/dart/ide/hierarchy/call/DartCalleeTreeStructure.java

```diff
@@ -0,0 +1,165 @@
+package com.jetbrains.lang.dart.ide.hierarchy.call;
+
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.vfs.LocalFileSystem;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiFile;
+import com.intellij.psi.util.PsiTreeUtil;
+import com.jetbrains.lang.dart.analyzer.DartAnalysisServerService;
+import com.jetbrains.lang.dart.psi.*;
+import org.dartlang.analysis.server.protocol.ElementKind;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.jetbrains.lang.dart.analyzer.DartServerData.DartNavigationRegion;
+import static com.jetbrains.lang.dart.analyzer.DartServerData.DartNavigationTarget;
+
+public class DartCalleeTreeStructure extends DartCallHierarchyTreeStructure {
+
+  public DartCalleeTreeStructure(Project project, PsiElement element, String currentScopeType) {
+    super(project, element, currentScopeType);
+  }
+
+  private static void getCallees(@NotNull PsiElement element, @NotNull List<PsiElement> results) {
+    DartComponentName name = (DartComponentName)element;
+    DartComponent decl = (DartComponent)name.getParent();
+    PsiFile file = PsiTreeUtil.getParentOfType(decl, PsiFile.class);
+    if (file == null) return;
+    VirtualFile vFile = file.getVirtualFile();
+    List<DartNavigationRegion> navRegions =
+      DartAnalysisServerService.getInstance().analysis_getNavigation(vFile, decl.getTextOffset(), decl.getTextLength());
+    if (navRegions == null) return;
+    resolveReferences(decl, navRegions, results);
+  }
+
+  private static void resolveReferences(@NotNull DartComponent component,
+                                        @NotNull List<DartNavigationRegion> regions,
+                                        @NotNull List<PsiElement> results) {
+    component.acceptChildren(new DartRecursiveVisitor() {
+      @Override
+      public void visitReferenceExpression(@NotNull DartReferenceExpression reference) {
+        int offset = reference.getTextOffset();
+        List<DartNavigationTarget> targets = getRegionAt(offset, regions);
+        for (DartNavigationTarget target : targets) {
+          if (isExecutable(target)) {
+            PsiElement element = getDeclaration(target, reference);
+            if (element != null) {
+              results.add(element);
+            }
+          }
+        }
+        super.visitReferenceExpression(reference);
+      }
+    });
+  }
+
+  private static boolean isExecutable(DartNavigationTarget target) {
+    String kind = target.getKind();
+    return ElementKind.METHOD.equals(kind) ||
+           ElementKind.FUNCTION.equals(kind) ||
+           ElementKind.GETTER.equals(kind) ||
+           ElementKind.SETTER.equals(kind) ||
+           ElementKind.CONSTRUCTOR.equals(kind);
+  }
+
+  private static PsiElement getDeclaration(DartNavigationTarget target, PsiElement reference) {
```

Comment:
> I don't understand this comment.

That comment was for the line where `DartCalleeTreeStructure.getDeclaration()` method is declared. The purpose of this method is to find `PsiElement` that corresponds to the `DartNavigationTarget` parameter. My comment is that there's already a method that performs this task, so it can be reused: `DartResolver.getElementForNavigationTarget()`.